src/project-css/app/Http/Controllers/DataViewController.php

```diff
@@ -91,4 +99,120 @@ public function show(Request $request, $curTable, $curId){
                             ->with('tblId',$curTable);
   }
 
+  public function search(Request $request, $curTable, $search = NULL, $page = 1){
+    // test for the validity of curtable
+    if(!$this->isValidTable($curTable)){
+      return redirect()->route('home')->withErrors(['Table id is invalid']);
+    }
+
+    // Get the table entry in meta table "tables"
+    $curTable = Table::find($curTable);
+    if(!$curTable->hasAccess){
+      return redirect()->route('home')->withErrors(['Table is disabled']);
+    }
+
+    if ($search == NULL){
+      $search = $request->input('search');
+    }
+
+    $string_helper = new customStringHelper();
+    $srchStrng = $string_helper->cleanSearchString($search);
+
+    // set records per page
+    $perPage = 30;
+
+    // retrieve the column names
+    $clmnNmes = DB::getSchemaBuilder()->getColumnListing($curTable->tblNme);
+
+    // query sorted by revelancy score
+    $query = DB::table($curTable->tblNme)
+            ->whereRaw("match(srchindex) against (? in boolean mode)", [$srchStrng])
+            ->orderBy('score', 'desc')
+            ->offset($page-1 * $perPage)
+            ->limit($perPage);
+
+    $rcrds = $query
+            ->get(['*', DB::raw("MATCH (srchindex) AGAINST ('".$srchStrng."') AS score")]);
+
+    $rcrdsCount = count($rcrds);
+
+    // if fulltext search yeilds no results try a basic like search
```

Comment:
if this is a production this needs to be removed, anything moving in to master should not have commented code.

Reply:
removed